src/test/java/net/minecraftforge/test/VillagerProfessionTest.java

```diff
@@ -0,0 +1,21 @@
+package net.minecraftforge.test;
+
+import net.minecraftforge.event.RegistryEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.Mod.EventBusSubscriber;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.common.registry.VillagerRegistry.VillagerCareer;
+import net.minecraftforge.fml.common.registry.VillagerRegistry.VillagerProfession;
+
+@Mod(modid = "professiontest", name = "ProfessionTest2000", version = "1.0", acceptableRemoteVersions = "*")
```

Comment:
no it shouldnt. The vanilla client *should* be able to connect to servers with this mod it would just have zombies/villagers with the wrong professions.